sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AADOAuth2UserService.java

```diff
@@ -115,20 +118,39 @@ public OidcUser loadUser(OidcUserRequest userRequest) throws OAuth2Authenticatio
         if (!properties.allowedGroupIdsConfigured() && !properties.allowedGroupNamesConfigured()) {
             return Collections.emptySet();
         }
-        Set<String> groups = Optional.of(accessToken)
-                                     .map(AbstractOAuth2Token::getTokenValue)
-                                     .map(graphClient::getGroupsFromGraph)
-                                     .orElseGet(Collections::emptySet);
-        if (properties.getUserGroup().getEnableFullList()) {
-            return groups;
+        Map<String, Set<String>> groups = getGroups(accessToken);
+
+        if (properties.getUserGroup().getEnableFullList() || isAllGroupId()) {
+            Set<String> allowedGroupsName = Arrays
+                .asList(properties.getUserGroup().getAllowedGroupNames())
+                .stream()
+                .flatMap(Collection::stream)
+                .filter(groups.get(GROUP_NAME)::contains)
```

Comment:
It's better to use local variable instead of call `groups.get(GROUP_NAME)` multiple times.